Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptSearch.java

```diff
@@ -55,29 +60,53 @@
      * @param req
      *            Holds HTTP request information
      * @return
+     * @throws IllegalAccessException
+     * @throws IllegalArgumentException
      */
     @RequestMapping(value = "rest/ConceptSearch", method = RequestMethod.GET, produces = "application/xml; charset=utf-8")
-    public @ResponseBody ResponseEntity<String> searchConcept(HttpServletRequest req) {
-        Map<String, String[]> queryParams = req.getParameterMap();
+    public @ResponseBody ResponseEntity<String> searchConcept(
+            @Valid ConceptSearchParameters conceptSearchParameters)
```

Comment:
It's a good idea to validate those parameters, but there doesn't seem to be a validator?